frontend/components/user-settings/user-settings.tsx

```diff
@@ -0,0 +1,298 @@
+                        <AvatarFallback>Placeholder</AvatarFallback>
+                        </Avatar>
+                        <div>
+                        <Input type="file" accept="image/*" onChange={handleProfilePictureChange} className="mb-2" />
+                        <Button variant="outline" onClick={handleDeleteProfilePicture}>Delete Picture</Button>
+                        </div>
+                    </div>
+                    <div>
+                        <Label htmlFor="username">Username</Label>
+                        <Input id="username" name="username" value={user.username} onChange={handleInputChange} />
+                    </div>
+                    <div>
+                        <Label htmlFor="email">Email</Label>
+                        <Input id="email" name="email" type="email" value={user.email} onChange={handleInputChange} />
+                    </div>
+                    <div>
+                        <Label htmlFor="skillLevel">Skill Level</Label>
+                        <Select value={user.skillLevel} onValueChange={handleSkillLevelChange}>
+                        <SelectTrigger id="skillLevel">
+                            <SelectValue placeholder="Select skill level" />
+                        </SelectTrigger>
+                        <SelectContent>
+                            <SelectItem value="Novice">Novice</SelectItem>
+                            <SelectItem value="Intermediate">Intermediate</SelectItem>
+                            <SelectItem value="Expert">Expert</SelectItem>
+                        </SelectContent>
+                        </Select>
+                    </div>
+                </CardContent>
+                <CardFooter>
+                <Button onClick={handleSaveChanges}>Save Changes</Button>
+                <Toaster />
+                </CardFooter>
+            </Card>
+            </TabsContent>
+            <TabsContent value="account">
+            <Card>
+                <CardHeader>
+                <CardTitle>Account Security</CardTitle>
+                <CardDescription>Manage your password and account settings.</CardDescription>
+                </CardHeader>
+                <CardContent className="space-y-4">
+                <div>
+                    <Label htmlFor="currentPassword">Current Password</Label>
+                    <Input id="currentPassword" type="password" />
+                </div>
+                <div>
+                    <Label htmlFor="newPassword">New Password</Label>
+                    <Input id="newPassword" type="password" />
+                </div>
+                <div>
+                    <Label htmlFor="confirmPassword">Confirm New Password</Label>
+                    <Input id="confirmPassword" type="password" />
+                </div>
+                </CardContent>
+                <CardFooter>
+                <Button>Change Password</Button>
+                </CardFooter>
+            </Card>
```

Comment:
Could we abstract this out to another component? similar with the other sections, we can then place the logic in there which will make the main page.tsx cleaner